sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -4851,6 +4892,348 @@ public void onTimer2(
+    //              .advanceWatermarkTo(new Instant(0))
+    //              .addElements(KV.of("hello", 37))
+    //              .advanceWatermarkTo(new Instant(0).plus(Duration.standardSeconds(1)))
+    //              .advanceWatermarkToInfinity();
+    //
+    //      PCollection<Integer> output = pipeline.apply(stream).apply(ParDo.of(fn));
+    //      PAssert.that(output).containsInAnyOrder(3, 42);
+    //      pipeline.run();
+    //    }
+    //
+    //    @Test
+    //    @Category({NeedsRunner.class, UsesTimersInParDo.class, UsesTestStream.class})
+    //    public void testSetProcessingTimerAfterClear() {
+    //      final String timerId = "processing-timer";
+    //
+    //      DoFn<KV<String, Integer>, Integer> fn =
+    //          new DoFn<KV<String, Integer>, Integer>() {
+    //
+    //            @TimerId(timerId)
+    //            private final TimerSpec spec = TimerSpecs.timer(TimeDomain.PROCESSING_TIME);
+    //
+    //            @ProcessElement
+    //            public void processElement(
+    //                @Element KV<String, Integer> e,
+    //                @TimerId(timerId) Timer timer,
+    //                OutputReceiver<Integer> r) {
+    //              timer.clear();
+    //              timer.offset(Duration.standardSeconds(1)).setRelative();
+    //              r.output(3);
+    //            }
+    //
+    //            @OnTimer(timerId)
+    //            public void onTimer(TimeDomain timeDomain, OutputReceiver<Integer> r) {
+    //              r.output(42);
+    //            }
+    //          };
+    //
+    //      TestStream<KV<String, Integer>> stream =
+    //          TestStream.create(KvCoder.of(StringUtf8Coder.of(), VarIntCoder.of()))
+    //              .addElements(KV.of("hello", 37), KV.of("hello", 38))
+    //              .advanceProcessingTime(
+    //                  Duration.millis(
+    //                          DateTimeUtils.currentTimeMillis() / 1000 * 1000) // round to seconds
+    //                      .plus(Duration.standardMinutes(2)))
+    //              .advanceWatermarkToInfinity();
+    //
+    //      PCollection<Integer> output = pipeline.apply(stream).apply(ParDo.of(fn));
+    //      PAssert.that(output).containsInAnyOrder(3, 3, 42);
+    //      pipeline.run();
+    //    }
+    //
+    //    @Test
+    //    @Category({NeedsRunner.class, UsesTimersInParDo.class, UsesTestStream.class})
+    //    public void testSetEventTimerAfterClear() {
+    //      final String timerId = "event-timer";
+    //
+    //      DoFn<KV<String, Integer>, Integer> fn =
+    //          new DoFn<KV<String, Integer>, Integer>() {
+    //
+    //            @TimerId(timerId)
+    //            private final TimerSpec spec = TimerSpecs.timer(TimeDomain.EVENT_TIME);
+    //
+    //            @ProcessElement
+    //            public void processElement(@TimerId(timerId) Timer timer, OutputReceiver<Integer>
```

Comment:
Did you mean to leave these large sections commented out?

Reply:
No I didn't. thanks for catching!